frontend/packages/ceph-storage-plugin/src/components/create-backingstore-page/backing-store-dropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Alert,
+  Dropdown,
+  DropdownItem,
+  DropdownSeparator,
+  DropdownToggle,
+} from '@patternfly/react-core';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { ListKind } from 'public/module/k8s';
+import CreateBackingStoreFormModal from './create-bs-modal';
+import { NooBaaBackingStoreModel } from '../../models';
+import { BackingStoreKind } from '../../types';
+
+export const BackingStoreDropdown: React.FC<BackingStoreDropdownProps> = ({
+  namespace,
+  onChange,
+  className,
+  selectedKey,
+}) => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = React.useState(false);
+  const [nbsObj, nbsLoaded, nbsErr] = useK8sGet<ListKind<BackingStoreKind>>(
+    NooBaaBackingStoreModel,
+    null,
+    namespace,
+  );
+  const [nsName, setNSName] = React.useState(selectedKey || '');
+  const [dropdownItems, setDropdownItems] = React.useState([]);
+  const nbsList = nbsLoaded ? nbsObj.items : [];
+  const handleDropdownChange = React.useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      setNSName(e.currentTarget.id);
+      onChange(nbsList.find((nbs) => nbs?.metadata?.name === e.currentTarget.id));
+    },
+    [nbsList, onChange],
+  );
+  React.useEffect(() => {
+    const nbsDropdownItems = nbsList.reduce(
```

Comment:
you can use useMemo , which will memoize the dropdown results for you,

Reply:
ok i have changed it to useMemo